src-graphql/client.js

```diff
@@ -24,11 +24,11 @@ function fetchAllProductResources(product, client) {
   const images = product.images;
   const variants = product.variants;
 
-  if (images && images[images.length - 1].hasNextPage.valueOf()) {
+  if (images && images.length && images[images.length - 1].hasNextPage.valueOf()) {
```

Comment:
Unrelated, but is `valueOf` needed for these? I thought `hasNextPage` is just a boolean?

edit:

```js
function hasNextPage(connection, edge) {
  if (edge !== connection.edges[connection.edges.length - 1]) {
    return true;
  }

  return connection.pageInfo.hasNextPage;
}
```

Maybe `valueOf` should be called in this function on `connection.pageInfo.hasNextPage` to be consistent.